bouncer/commands/new_swap.ts

```diff
@@ -1,35 +1,97 @@
 
+import yargs from 'yargs';
+import { hideBin } from 'yargs/helpers';
 import { InternalAsset } from '@chainflip/cli';
 import { parseAssetString, executeWithTimeout } from '../shared/utils';
 import { requestNewSwap } from '../shared/perform_swap';
+import { RefundParameters } from '../shared/new_swap';
+
+interface Args {
+  sourceAsset: string;
+  destAsset: string;
+  destAddress: string;
+  maxBoostFeeBps: number;
+  refundAddress?: string;
+  minPrice?: string;
+  refundDuration: number;
+}
 
 async function newSwapCommand() {
-  const sourceAsset = parseAssetString(process.argv[2]);
-  const destAsset = parseAssetString(process.argv[3]);
-  const destAddress = process.argv[4];
-  const maxBoostFeeBps = Number(process.argv[5] || 0);
+  const args = yargs(hideBin(process.argv))
+    .command(
+      '$0 <sourceAsset> <destAsset> <destAddress> [maxBoostFeeBps] [refundAddress] [minPrice] [refundDuration]',
+      'Request a new swap with the provided parameters',
+      (a) => {
+        console.log('Parsing options');
+        return a
+          .positional('sourceAsset', {
+            describe: 'The source currency ("Btc", "Eth", "Dot", "Usdc")',
+            type: 'string',
+          })
+          .positional('destAsset', {
+            describe: 'The destination currency ("Btc", "Eth", "Dot", "Usdc")',
+            type: 'string',
+          })
+          .positional('destAddress', {
+            describe: 'The destination address',
+            type: 'string',
+          })
+          .option('maxBoostFeeBps', {
+            describe: 'The max boost fee bps (default: 0 (no boosting))',
+            type: 'number',
+            default: 0,
+            demandOption: false,
+          })
+          .option('refundAddress', {
+            describe: 'Fill or Kill refund address',
+            type: 'string',
+            demandOption: false,
+          })
+          .option('minPrice', {
+            describe: 'Fill or Kill minimum price',
+            type: 'string',
+            demandOption: false,
+          })
+          .option('refundDuration', {
+            describe: 'Fill or kill duration in blocks to retry swap before refunding',
+            type: 'number',
+            demandOption: false,
+            default: 0,
+          });
+      },
+    )
+    .help('h').argv as unknown as Args;
 
-  console.log(`Requesting swap ${sourceAsset} -> ${destAsset}`);
+  if ((args.refundAddress === undefined) !== (args.minPrice === undefined)) {
+    throw new Error('Must specify both refundAddress and minimumPrice when using refund options');
+  }
+  const refundParameters: RefundParameters | undefined =
+    args.refundAddress !== undefined
+      ? {
+          retryDurationBlocks: args.refundDuration,
+          refundAddress: args.refundAddress,
+          minPrice: args.minPrice ? args.minPrice : '0',
```

Comment:
we could just use default as `0` instead of this?

Reply:
This '0' will never get hit because of the check above. Just needed this to make the linter happy. I can change it to something like 'error' so its more obvious.